kufar/books/main.go

```diff
@@ -205,27 +338,41 @@ func (d Database) UpdateBook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	bk := Book{}
-
+	var bk Book
 	if err := json.NewDecoder(r.Body).Decode(&bk); err != nil {
-		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		http.Error(w, http.StatusText(http.StatusInternalServerError)+": Error unmarshalling json", http.StatusInternalServerError)
+		return
+	}
+	
+	realm := "Access to the users private books"
```

Comment:
You mixed up basic auth stuff with sessions :(
Let's remove it for now and tomorrow we will have a talk and try to integrate basic auth in our scheme